// @ionic-native/rover/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
/**
 * This is a template for new plugin wrappers
 *
 * TODO:
 * - Add/Change information below
 * - Document usage (importing, executing main functionality)
 * - Remove any imports that you are not using
 * - Add this file to /src/index.ts (follow style of other plugins)
 * - Remove all the comments included in this template, EXCEPT the @Plugin wrapper docs and any other docs you added
 * - Remove this note
 *
 */
import { Injectable } from '@angular/core';
import { Plugin, Cordova, IonicNativePlugin } from '@ionic-native/core';
// import { Plugin, Cordova, CordovaProperty, CordovaInstance, InstanceProperty, IonicNativePlugin } from '@ionic-native/core';
// import { Observable } from 'rxjs/Observable';
/**
 * @name Rover
 * @description
 * Wrapper to call the Cordova Rover from Ionic
 *
 * @usage
 * ```typescript
 * import { Rover } from '@ionic-native/rover';
 *
 *
 * constructor(private Rover: Rover) { }
 *
 * ...
 *
 * ```
 */
var Rover = (function (_super) {
    __extends(Rover, _super);
    function Rover() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * Start the GPS tracking
     * <@param arg1 {string} Some param to configure something>
     * @return {Promise<any>} Returns a promise when succesful.
     */
    Rover.prototype.startTracking = function () {
        return; // We add return; here to avoid any IDE / Compiler errors
    };
    /**
     * Stop the GPS tracking
     * @return {Promise<any>} Returns a promise when succesful.
     */
    Rover.prototype.stopTracking = function () {
        return; // We add return; here to avoid any IDE / Compiler errors
    };
    return Rover;
}(IonicNativePlugin));
Rover.decorators = [
    { type: Injectable },
];
/** @nocollapse */
Rover.ctorParameters = function () { return []; };
__decorate([
    Cordova(),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", Promise)
], Rover.prototype, "startTracking", null);
__decorate([
    Cordova(),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", Promise)
], Rover.prototype, "stopTracking", null);
Rover = __decorate([
    Plugin({
        pluginName: 'Rover',
        plugin: 'cordova-plugin-rover',
        pluginRef: 'navigator.rover',
        platforms: ['Android'] // Array of platforms supported, example: ['Android', 'iOS']
    })
], Rover);
export { Rover };
